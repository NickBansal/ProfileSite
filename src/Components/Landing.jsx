import React, { Component } from 'react'
import Mountain from '../Stylesheets/Images/Mountain.JPG'
import Me from '../Stylesheets/Images/Me.JPG'
import '../Stylesheets/Landing.css'

class Navbar extends Component {

    state = {
        open: false
    }

    render() {
        const style = {
            backgroundImage: `url(${Mountain})`,
            minHeight: '100vh',
            top: '0',
            left: '0',
            width: '100%',
            height: '100%',
            backgroundSize: 'cover',
            zIndex: '-2',
            backgroundPosition: 'center'
        }
        return (
            <div style={style} className="TitlePage">
                <nav>
                    <h1 id="Name">Nick Bansal</h1>
                    <ul className="NavLinks">
                        <li>About</li>
                        <li>Projects</li>
                        <li>Contact</li>
                    </ul>
                    <div 
                    onClick={this.handleClick}
                    className="Hamburger">
                        <div className={this.state.open ? "burger-menu open" : "burger-menu"}>
                            <div className="bar1" key="b1" />
                            <div className="bar2" key="b2" />
                            <div className="bar3" key="b3" />
                        </div>
                    </div>
                </nav>
                <h3>Author, DreamWeaver, Visionary...</h3>
                <h1 id="MeJob">Full Stack Developer</h1>
                <div className="ContactIcons">
                    <i class="fab fa-github fa-3x"></i>
                    <i class="fab fa-stack-overflow fa-3x"></i>
                    <i class="fab fa-codepen fa-3x"></i>
                    <i class="fas fa-code fa-3x"></i>
                </div>
                <img id="MePic" src={Me} alt="Me"/>
            </div>
        )
    }

    handleClick = () => {
        this.setState({
            open: !this.state.open
        })
    }
}

export default Navbar