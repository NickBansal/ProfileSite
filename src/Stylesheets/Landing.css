/* MY DESCRIPTION */

h3 {
  color: white;
  font-size: 30px;
  opacity: 0;
  text-shadow: 1px 1px 7px black;
  -webkit-animation: 1 fadein 1s forwards;
          animation: 1 fadein 1s forwards;
}


#MeJob {
  font-size: 50px;
  text-shadow: 1px 1px 7px black;
  color: white;
  margin-right: 450px;
  opacity: 0;
  -webkit-animation: 1 titleSlide 1s forwards;
          animation: 1 titleSlide 1s forwards;
  -webkit-animation-delay: 0.5s;
          animation-delay: 0.5s
}

#MePic {
  width: 270px;
  border-radius: 120px;
  border: 5px solid white;
  margin-left: 450px;
  opacity: 0;
  -webkit-animation: 1 picSlide 1s forwards;
          animation: 1 picSlide 1s forwards;
  -webkit-animation-delay: 0.5s;
          animation-delay: 0.5s;
  -webkit-box-shadow: none;
          box-shadow: none
}

@-webkit-keyframes titleSlide {
  100% {
    opacity: 1;
    margin-right: 0
  }
}

@keyframes titleSlide {
  100% {
    opacity: 1;
    margin-right: 0
  }
}

@-webkit-keyframes picSlide {
  100% {
    opacity: 1;
    margin-left: 0;
    -webkit-box-shadow: -1px 4px 40px 1px white ;
    box-shadow: -1px 4px 40px 1px white;
  }
}

@keyframes picSlide {
  100% {
    opacity: 1;
    margin-left: 0;
    -webkit-box-shadow: -1px 4px 40px 1px white ;
    box-shadow: -1px 4px 40px 1px white;
  }
}

@media (max-width: 650px) {
  #MeJob {
    font-size: 35px;
  }

  #MePic {
    width: 250px;
  }

  h3 {
    font-size: 20px;
  }

  li {
    margin: 0 20px
  }

  nav {
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-orient: vertical;
    -webkit-box-direction: normal;
        -ms-flex-direction: column;
            flex-direction: column
  }
}