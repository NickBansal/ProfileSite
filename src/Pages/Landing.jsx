import React from 'react'
import Mountain from '../Stylesheets/Images/Mountain.JPG'
import Me from '../Stylesheets/Images/Me.JPG'
import IconLinks from '../Components/IconLinks'
import Snowflakes from '../Components/Snowflakes'
import DownArrow from '../Components/DownArrow'

import '../Stylesheets/Landing.css'

const Navbar = () => {
    const style = {
        backgroundImage: `url(${Mountain})`,
        minHeight: '100vh',
        top: '0',
        left: '0',
        width: '100%',
        height: '100%',
        backgroundSize: 'cover',
        backgroundAttachment: 'fixed',
        zIndex: '-2',
        backgroundPosition: 'center'
    }


    return (
        <div style={style} className="TitlePage">
            <nav>
                <h1 id="Name">Nick Bansal</h1>
                <ul className="NavLinks">
                    <li>About</li>
                    <li>Projects</li>
                    <li>Contact</li>
                </ul>
            </nav>
            <div style={{ paddingTop: '100px' }}>
                <h3>Author, DreamWeaver, Visionary...</h3>
                <h1 id="MeJob">Full Stack Developer</h1>
                <IconLinks />
                <Snowflakes />
                <img id="MePic" src={Me} alt="Me" />
            </div>
            <DownArrow />
        </div>
    )
}

export default Navbar