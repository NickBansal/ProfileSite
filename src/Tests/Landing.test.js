import './setupTest';
import React from 'react';
import Landing from '../Pages/Landing';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'

describe('<Landing />', () => {
    let wrapper
    beforeEach(() => {
        wrapper = shallow(<Landing />)
    }) 
    it('Component matches the snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot()
    })
    it('Checks that the main div component with id TitlePage exists', () => {
        expect(wrapper.find('div#TitlePage').exists()).toBe(true)
    })
    it('Counts the amount of divs on the page', () => {
        expect(wrapper.find('div').length).toBe(2)
    })
})