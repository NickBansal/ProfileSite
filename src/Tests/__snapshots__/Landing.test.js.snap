// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Landing /> Component matches the snapshot 1`] = `
<div
  id="TitlePage"
  style={
    Object {
      "backgroundAttachment": "fixed",
      "backgroundImage": "url(Mountain.JPG)",
      "backgroundPosition": "center",
      "backgroundSize": "cover",
      "height": "100%",
      "left": "0",
      "minHeight": "100vh",
      "top": "0",
      "width": "100%",
      "zIndex": "-2",
    }
  }
>
  <Navbar />
  <div
    style={
      Object {
        "paddingTop": "100px",
      }
    }
  >
    <h3>
      Author, DreamWeaver, Visionary...
    </h3>
    <h1
      id="MeJob"
    >
      Full Stack Developer
    </h1>
    <IconLinks />
    <SnowFlakes />
    <img
      alt="Me"
      id="MePic"
      src="Me.JPG"
    />
  </div>
  <DownArrow />
  <Footer />
</div>
`;
